sycl/source/detail/pi.cpp

```diff
@@ -131,6 +150,55 @@ vector_class<plugin> initialize() {
   // TODO: Correct the logic to store the appropriate plugin into GlobalPlugin
   // variable. Currently it saves the last plugin found.
   GlobalPlugin = std::make_shared<plugin>(PluginInformation);
+
+#ifdef XPTI_ENABLE_INSTRUMENTATION
+  // Not sure this is the best place to initialize
```

Comment:
I believe this "initialize" function  can be called multiple times. @garimagu Could you please  confirm?

Reply:
Addressed this with a boolean state variable